trunk/source/TraceAbstraction/src/de/uni_freiburg/informatik/ultimate/plugins/generator/traceabstraction/NwaCegarLoop.java

```diff
@@ -542,6 +545,23 @@ protected void computeIcfgHoareAnnotation() {
 					getServices(), mSimplificationTechnique, mXnfConversionTechnique);
 			writer.addHoareAnnotationToCFG();
 			mCegarLoopBenchmark.addHoareAnnotationData(clha.getHoareAnnotationStatisticsGenerator());
+
+			// Temporary workaround: print invariants in every CEGAR loop
+			final IBacktranslationService backTranslatorService = mServices.getBacktranslationService();
+			final Set<IcfgLocation> locsForLoopLocations = new HashSet<>();
+			locsForLoopLocations.addAll(IcfgUtils.getPotentialCycleProgramPoints(mIcfg));
+			locsForLoopLocations.addAll(mIcfg.getLoopLocations());
+			// find all locations that have outgoing edges which are annotated with LoopEntry, i.e., all loop candidates
+
+			for (final var entry : clha.getLoc2hoare().entrySet()) {
+				final Term formula = entry.getValue().getFormula();
+				final InvariantResult<IIcfgElement, Term> invResult =
+						new InvariantResult<>(Activator.PLUGIN_NAME, entry.getKey(), backTranslatorService, formula);
+				// TODO .reportResult(invResult);
+				mLogger.warn("invariant at location %s :: %s", entry.getKey(), formula);
+				mLogger.warn(invResult);
+			}
```

Comment:
Note: I hope to get `wip/dk/proof-refactoring` merged next week, which will allow us to avoid this hack and output separate proofs for different CEGAR loops in a nicer fashion.

Reply:
@maul-esel 
I talked with @henkele a little bit about this problem and I don't see an easy way to have separate invariants per error location if using one one CEGAR. 
The only thing I could imagine to be somewhat helpful:
* Keep a mapping from error location to Cegar iteration and record in which iteration `i` a error location was first marked as unreachable. 
* Only use the interpolant automata from iterations `0..i-1` to generate invariants for that particular error location.

Of course in most cases, these invariants are stronger than necessary, but I guess its easily implemented and might provide some insight for better ways.

Reply:
> I talked with @henkele a little bit about this problem and I don't see an easy way to have separate invariants per error location if using one one CEGAR.

Oh sure, that would be much more difficult. Both the workaround on this branch, and the support in #671 are for the case the we have separate CEGAR loops for each error location.